examples/graph_cross_network.py

```diff
@@ -0,0 +1,248 @@
+        return self.graphs[idx]
+
+    def statistics(self):
+        return self.graphs[0].x.shape[1], self.num_classes
+
+
+class GraphClassifier(torch.nn.Module):
+    """
+    Description
+    -----------
+    Graph Classifier for graph classification.
+    GXN + MLP
+    """
+    def __init__(self, args):
+        super(GraphClassifier, self).__init__()
+        self.gxn = GraphCrossNet(in_dim=args.in_dim,
+                                 out_dim=args.embed_dim,
+                                 edge_feat_dim=args.edge_feat_dim,
+                                 hidden_dim=args.hidden_dim,
+                                 pool_ratios=args.pool_ratios,
+                                 readout_nodes=args.readout_nodes,
+                                 conv1d_dims=args.conv1d_dims,
+                                 conv1d_kws=args.conv1d_kws,
+                                 cross_weight=args.cross_weight,
+                                 fuse_weight=args.fuse_weight)
+        self.lin1 = torch.nn.Linear(args.embed_dim, args.final_hidden_dim)
+        self.lin2 = torch.nn.Linear(args.final_hidden_dim, args.out_dim)
+        self.dropout = args.dropout
+
+    def forward(self, graph):
+        embed, logits1, logits2 = self.gxn(graph)
+        logits = F.relu(self.lin1(embed))
+        if self.dropout > 0:
+            logits = F.dropout(logits, p=self.dropout, training=self.training)
+        logits = self.lin2(logits)
+        return F.log_softmax(logits, dim=1), logits1, logits2
+
+
+def compute_loss(cls_logits: Tensor, labels: Tensor, logits_s1: Tensor,
+                 logits_s2: Tensor, epoch: int, total_epochs: int,
+                 device: torch.device):
+    # classification loss
+    classify_loss = F.nll_loss(cls_logits, labels.to(device))
+
+    # loss for vertex infomax pooling
+    scale1, scale2 = logits_s1.size(0) // 2, logits_s2.size(0) // 2
+    s1_label_t, s1_label_f = torch.ones(scale1), torch.zeros(scale1)
+    s2_label_t, s2_label_f = torch.ones(scale2), torch.zeros(scale2)
+    s1_label = torch.cat((s1_label_t, s1_label_f), dim=0).to(device)
+    s2_label = torch.cat((s2_label_t, s2_label_f), dim=0).to(device)
+
+    pool_loss_s1 = F.binary_cross_entropy_with_logits(logits_s1, s1_label)
+    pool_loss_s2 = F.binary_cross_entropy_with_logits(logits_s2, s2_label)
+    pool_loss = (pool_loss_s1 + pool_loss_s2) / 2
+
+    loss = classify_loss + (2 - epoch / total_epochs) * pool_loss
+
+    return loss
+
+
+def train(model: torch.nn.Module, optimizer, trainloader,
+          device, curr_epoch, total_epochs):
+    model.train()
+
+    total_loss = 0.
+    num_batches = len(trainloader)
+
+    for batch in trainloader:
+        optimizer.zero_grad()
+        batch.to(device)
+        out, l1, l2 = model(batch)
+        loss = compute_loss(out, batch.y.long(), l1, l2, curr_epoch,
+                            total_epochs, device)
+        loss.backward()
+        optimizer.step()
+
+        total_loss += loss.item()
```

Comment:
Let's compute that similar to [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/mutag_gin.py#L76) to account for the last batch size correctly.